include/drivers/bluetooth/hci_driver.h

```diff
@@ -185,6 +185,16 @@ struct bt_hci_driver {
 	 * @return 0 on success or negative error number on failure.
 	 */
 	int (*send)(struct net_buf *buf);
+
+	/**
+	 * @brief HCI vendor-specific setup
+	 *
+	 * Executes vendor-specific commands sequence to initialize
+	 * BT Controller before BT Host executes Reset sequence.
+	 *
+	 * @return 0 on success or negative error number on failure.
+	 */
+	int (*setup)(void);
```

Comment:
This needs to have `#if defined(BT_HCI_SETUP)` guards now that the Kconfig option applies to the HCI driver interface rather than just the H4 driver. Otherwise this PR is starting to look ready to be merged.